### PR TITLE
Add Polish to Virtual Keyboard

### PR DESCRIPTION
**Pull request by ihhub**
Polish has extra 9 characters on top of English alphabet.
image

relates to #6924

### CODE REVIEW

File: src/fheroes2/gui/ui_keyboard.cpp [hunk 1]
```diff
@@ -79,27 +97,52 @@ namespace
...
+            _window.reset();
+
+            _window = std::make_unique<fheroes2::StandardWindow>( offset.x, offset.y, size.width, size.height, true, _output );
+
+            _window->render();
 
             renderInputArea();
 
-            _output.render( getWindowRoi() );
+            _output.render( _window->activeArea() );
```
Comment (Districh-ru) [inline src/fheroes2/gui/ui_keyboard.cpp:145]: While resizing the keyboard window we do **render()** in `renderInputArea()` then here and then we add buttons and do a third render in `processVirtualKeyboardEvent()`.
So we get an extra frame without buttons and sometimes partially rendered:

https://github.com/ihhub/fheroes2/assets/113276641/634be282-2e56-4eea-9ebd-c13346018930

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Hi, @ihhub. I noticed one bug. Could you take a look when you have time.

File: src/fheroes2/gui/ui_keyboard.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/gui/ui_keyboard.cpp]: Hi @Districh-ru , could you please check the latest commit? I think it should be fixed. I cannot reproduce it properly as my machine is to powerful for this :(

Comment (ihhub) [reply src/fheroes2/gui/ui_keyboard.cpp]: @Districh-ru , on top of this I fixed another place where we were calling rendering for no reason.

Comment (Districh-ru) [reply src/fheroes2/gui/ui_keyboard.cpp]: @ihhub
I checked the fix. Now there are no rendering bugs. 👍

**Review by Districh-ru — APPROVED** [review]
The Polish Keyboard works fine.

I have an idea to discuss:
Since our Polish virtual keyboard has the same 3 rows as the English one, it might be good to show it by default. And also there is no need to switch back to English as all English letters are present in the Polish keyboard.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by Districh-ru** [comment]
Added diacritics to CP1250 characters used in Polish keyboard.
изображение

**Comment by Branikolog** [comment]
Hi, @ihhub 

> The Polish Keyboard works fine.
> 
> I have an idea to discuss: Since our Polish virtual keyboard has the same 3 rows as the English one, it might be good to show it by default. And also there is no need to switch back to English as all English letters are present in the Polish keyboard.

That's a reasonable point from @Districh-ru 

This would help to avoid texture floating behind the buttons, when we switch between languages and the window becomes larger.
Also, in this case the keyboard itself could be aligned in the center of the screen, since now it is not and I'm not sure we should do this with two keyboards available.

**Comment by ihhub** [comment]
@Districh-ru  and @Branikolog , we can do this outside this PR. Also don't forget about consistency which is important if we handle all other languages.

**Comment by Districh-ru** [comment]
> @Districh-ru and @Branikolog , we can do this outside this PR. Also don't forget about consistency which is important if we handle all other languages.

Yes, we can do this in other PR when we have more suppurted keyboard languages and we can decide how to behave with the ones that have all English letters in their alphabet. I suggested out this idea as it is noticeable here that switching to Polish keyboard simply adds the upper row without changing others.

**Review by Branikolog — APPROVED** [review]

**Pull request merged**